library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/callback/RecorderFragmentLifecycleCallback.kt

```diff
@@ -11,31 +11,33 @@ import androidx.fragment.app.DialogFragment
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import androidx.fragment.app.FragmentManager.FragmentLifecycleCallbacks
-import com.datadog.android.sessionreplay.internal.recorder.WindowCallbackInterceptor
 
 internal class RecorderFragmentLifecycleCallback(
-    private val windowCallbackInterceptor: WindowCallbackInterceptor
-) : FragmentLifecycleCallbacks() {
+    private val onWindowRefreshedCallback: OnWindowRefreshedCallback
+) :
+    FragmentLifecycleCallbacks() {
 
     override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
         super.onFragmentResumed(fm, f)
-        f.asValidDialogFragment {
-            val ownerActivity = it.dialog?.ownerActivity ?: return@asValidDialogFragment
-            it.getWindowsToRecord()?.let { windows ->
-                windowCallbackInterceptor.intercept(windows, ownerActivity)
-            }
+        getWindowsToRecord(f)?.let {
+            onWindowRefreshedCallback.onWindowsAdded(it)
         }
     }
 
     override fun onFragmentPaused(fm: FragmentManager, f: Fragment) {
-        f.asValidDialogFragment {
-            it.getWindowsToRecord()?.let { windows ->
-                windowCallbackInterceptor.stopIntercepting(windows)
-            }
+        getWindowsToRecord(f)?.let {
+            onWindowRefreshedCallback.onWindowsRemoved(it)
         }
         super.onFragmentPaused(fm, f)
     }
 
+    private fun getWindowsToRecord(f: Fragment): List<Window>? {
```

Comment:
I am actually not sure about this as I want to avoid calling the callback anyway if there is no new valid window added to avoid too many calls in the MainThread stack.